README.md
# NotifyModuleV2
The branch is for Community Driven Stable-Only updates.

Features added that was not before:
~Custom colors can be used by sending a table of color values like so:
```
  local notify = require(game.Workspace:WaitForChild("Services"):FindFirstChild("notify")) -- Replace "notify" with the location of the module
  --local color = {["R"]=255, ["G"]=0, ["B"]=0,}
  notify.notify(Player,"TitleCard", "Message", color, DLY)
```

along with that the Tween service was rewritten, so now upon providing ANY delay Notify will use a seperate script to Tween
Allowing the main script(Notify Module) to run without a delay, rather than waiting for the tween service to finish fully

The reason for the tween rewrite was during a Playeradded function would take ages to finish because it has to finish the tween for every player
This workaround allows anyone that uses this verison of notify to call 6 notifactions side-by-side and still run the scripts like datastore loading, or Charachter loading

Original.MD will contain the original Code from simplyNotifiedV1
Changes.MD will contain what i changes i have made in simplyNotifiedV1



Changes.md
```
local module = {}
local TweenService = game.Workspace:WaitForChild("Services"):FindFirstChild("notify"):FindFirstChild("TweenService"):FindFirstChild("Event") -- Replace "notify" with the location of the module
local NVersion=script:WaitForChild("Version")
NVersion.Value="V1.02"
print("Started notify Module")
function insertContainer(p)
	if p == nil then
		local simpleSuite = Instance.new("ScreenGui")
		simpleSuite.Name = "simpleSuite"
		simpleSuite.DisplayOrder = 25
		simpleSuite.ZIndexBehavior = Enum.ZIndexBehavior.Sibling
		simpleSuite.Parent = game:GetService("StarterGui")

		local Notifications = Instance.new("Frame")
		Notifications.BackgroundTransparency = 1
		Notifications.Name = "Notifications"
		Notifications.Position = UDim2.new(0.815, 0, 0.08, 0)
		Notifications.Size = UDim2.new(0.175, 0, 0.9, 0)
		Notifications.Parent = simpleSuite

		local UIListLayout = Instance.new("UIListLayout")
		UIListLayout.HorizontalAlignment = Enum.HorizontalAlignment.Center
		UIListLayout.Padding = UDim.new(0.025, 0)
		UIListLayout.VerticalAlignment = Enum.VerticalAlignment.Bottom
		UIListLayout.Parent = Notifications
	else
		local simpleSuite = Instance.new("ScreenGui")
		simpleSuite.Name = "simpleSuite"
		simpleSuite.DisplayOrder = 25
		simpleSuite.ZIndexBehavior = Enum.ZIndexBehavior.Sibling
		simpleSuite.Parent = p:WaitForChild("PlayerGui")

		local Notifications = Instance.new("Frame")
		Notifications.BackgroundTransparency = 1
		Notifications.Name = "Notifications"
		Notifications.Position = UDim2.new(0.815, 0, 0.08, 0)
		Notifications.Size = UDim2.new(0.175, 0, 0.9, 0)
		Notifications.Parent = simpleSuite

		local UIListLayout = Instance.new("UIListLayout")
		UIListLayout.HorizontalAlignment = Enum.HorizontalAlignment.Center
		UIListLayout.Padding = UDim.new(0.025, 0)
		UIListLayout.VerticalAlignment = Enum.VerticalAlignment.Bottom
		UIListLayout.Parent = Notifications
	end
end

insertContainer()

local colors = {
	NILColor = Color3.fromRGB(79, 79, 79),
	alert = Color3.fromRGB(255, 105, 105),
	success = Color3.fromRGB(105, 255, 105),
	warning = Color3.fromRGB(255, 255, 105),
	info = Color3.fromRGB(105, 105, 255),
	default = Color3.fromRGB(255, 255, 255)	
}

function module.notify(target,title,text,color,dur)

	if target:WaitForChild("PlayerGui"):FindFirstChild("simpleSuite") ~= nil then

		local duration

		if dur == nil then
			duration = 3
		else if dur ~= nil and dur >= 0.1 then
				duration = tonumber(dur)
			end
		end
		--Rescripted by botwot7024 (Below this line)
		if title ~= nil and text ~= nil and color ~= nil then
			local selectedColor = colors.default
			if type(color)=="string" then

				if string.lower(color) == "alert" or string.lower(color) == "success" or string.lower(color) == "warning" or string.lower(color) == "info" or string.lower(color) == "default" then

					if string.lower(color) == "alert" then
						selectedColor = colors.alert
					end

					if string.lower(color) == "success" then
						selectedColor = colors.success
					end

					if string.lower(color) == "warning" then
						selectedColor = colors.warning
					end
					if string.lower(color) == "info" then
						selectedColor = colors.info
					end
					if string.lower(color) == "default" then
						selectedColor = colors.default
					end
				else
					selectedColor = colors.NILColor
				end	

				
			elseif type(color)=="table" then
				--The New DefaultColor System
				if color.R==nil or color.G==nil or color.B==nil then
					selectedColor = colors.NILColor
				else
					selectedColor = Color3.fromRGB(math.clamp(color.R*255,0,255), math.clamp(color.G*255,0,255), math.clamp(color.B*255,0,255))
				end
			else--if no color defined
				selectedColor = colors.NILColor
			end
			--Rescripted by botwot7024 (Above this line)
			local notification_number = 1


			if game.Players:FindFirstChild(target.Name) then
				local player = game.Players:FindFirstChild(target.Name)

				for _,v in pairs(player.PlayerGui.simpleSuite.Notifications:GetChildren()) do

					notification_number = notification_number +1

				end


				local Notification1 = Instance.new("Frame")
				Notification1.BackgroundColor3 = Color3.fromRGB(25, 25, 25)
				Notification1.BorderSizePixel = 0
				Notification1.Name = "Notification"..notification_number
				Notification1.Size = UDim2.new(0, 0, 0, 0)
				Notification1.Parent = player.PlayerGui:WaitForChild("simpleSuite").Notifications

				local Color = Instance.new("Frame")
				Color.BackgroundColor3 = selectedColor
				Color.BorderSizePixel = 0
				Color.Name = "Color"
				Color.Size = UDim2.new(0.025, 0, 1, 0)
				Color.Parent = Notification1

				local Content = Instance.new("Frame")
				Content.BackgroundTransparency = 1
				Content.Name = "Content"
				Content.Position = UDim2.new(0.025, 0, 0, 0)
				Content.Size = UDim2.new(0.975, 0, 1, 0)
				Content.Parent = Notification1

				local Title = Instance.new("TextLabel")
				Title.BackgroundTransparency = 1
				Title.Font = Enum.Font.SourceSansSemibold
				Title.Name = "Title"
				Title.Size = UDim2.new(1, 0, 0.4, 0)
				Title.Text = title
				Title.TextColor3 = Color3.fromRGB(255, 255, 255)
				Title.TextScaled = true
				Title.TextSize = 14
				Title.TextWrapped = true
				Title.Parent = Content

				local Text = Instance.new("TextLabel")
				Text.BackgroundTransparency = 1
				Text.Font = Enum.Font.SourceSans
				Text.Name = "Text"
				Text.Position = UDim2.new(0, 0, 0.2999997, 0)
				Text.Size = UDim2.new(1, 0, 0.7, 0)
				Text.Text = text
				Text.TextColor3 = Color3.fromRGB(255, 255, 255)
				Text.TextScaled = true
				Text.TextSize = 14
				Text.TextWrapped = true
				Text.Parent = Content

				--Rescripted by botwot7024
				if script:FindFirstChild("TweenService") then
					local Time=os.clock()
					print("Tweening 1 at "..Time)
					TweenService:Fire(Notification1, UDim2.new(1, 0, 0.1, 0), duration, UDim2.new(0, 0, 0, 0), 1.1, Time)
					print("Tweening 2 at "..os.clock())
				else

Updater.lua
--Settings
local ExpirmentalUpdates=true
local UseCustomVersion=false

--Script Start
if UseCustomVersion then
	ExpirmentalUpdates=false
end
if ExpirmentalUpdates then
	UseCustomVersion=false
end
local TweenService = game.Workspace:WaitForChild("Services"):FindFirstChild("notify"):FindFirstChild("TweenService"):FindFirstChild("Event") -- Replace "notify" with the location of the module
local NVersion=script.Parent:WaitForChild("Version")
--Gather Services
local CodeService = game:GetService("HttpService")
local NewCode = CodeService:GetAsync("https://raw.githubusercontent.com/Coaxgames/NotifyModuleV1-Reworked/CommunityRelease/Changes.md")
local NewInst = CodeService:GetAsync("https://raw.githubusercontent.com/Coaxgames/NotifyModuleV1-Reworked/CommunityRelease/InstuctionForModule.md")
--print(NewCode)
--Get local Notify Version
local Var1=script.Parent.Version.Value
local SplitV1=string.split(Var1, "V")
print(SplitV1)
local LocalVersion=SplitV1[2]
local FindVersion--GitVersion
LocalVersion=tonumber(LocalVersion)
if LocalVersion==nil then
	LocalVersion=0x001
end
--Look for version inside of NewCode to see if we are up to date
if NewCode~=nil then
	FindVersion=string.match(NewCode, "NVersion.Value="..'%b""')
	FindVersion=string.gsub(FindVersion, "NVersion.Value=", "")
	FindVersion=string.gsub(FindVersion, "\"", "")
	FindVersion=string.gsub(FindVersion, "V", "")
	FindVersion=tonumber(FindVersion)
	print("Local Version="..LocalVersion..", GitVersion="..FindVersion)
	if LocalVersion~=nil and LocalVersion < FindVersion then
		if ExpirmentalUpdates then
			print("Requesting Live Patch...")
			if game:GetService("ServerScriptService"):FindFirstChild("NotifyUpdater"):FindFirstChild("UpdateComs") then
				game:GetService("ServerScriptService").NotifyUpdater.UpdateComs:Fire(script.Parent.Parent:FindFirstChild("notify"), LocalVersion, FindVersion, NewCode, NewInst)
				return	
			else
				print("Cannot Patch: Missing UpdateComs Event")
			end
			print("Outdated Version, Please Update with NotifyUpdater...")
			warn("A new Update is available for NotifyV"..LocalVersion.." --> V"..FindVersion.."\nUse NotifyUpdater if you have it")
			if UseCustomVersion and script.Parent:FindFirstChild("notify(backup)") then
				print("Using Backup notify")
				--Remove this and use the backup Notify module
				script.Parent.Name="notnotify" --Rename this script
				for i, obj in pairs(script.Parent:GetChildren()) do--Move children to BackupModule
					if obj.Name~=script.Name then
						obj.Parent=script.Parent["notify(Backup)"]
					end
				end
				script.Parent["notify(Backup)"].Parent=script.Parent.Parent--re-parent main
				script.Parent.Parent["notify(Backup)"].Name="notify"--Rename Backup to notify
				print("Finished Loading notify")
				script.Parent=script.Parent.Parent["notify(Backup)"]
				script.Parent.Parent:FindFirstChild("notify"):Destroy()
				script:Destroy()--remove this script
			elseif UseCustomVersion and not script.Parent:FindFirstChild("notify(backup)") then
				print("Cannot Use Backup notify: Missing -notify(backup)-")
			else
				print("Backup Disabled")
			end

		else
			print("Updates are turned off, notify is out of date")
		end
	else
		print("up-To-Date Version, Removing Updater...")
	end
end

--StartUp




Instuctions.lua
if script.Enabled then--Turn off script if enabled for whatever reason
	script.Enabled=false
end
--[[
Modified By BotWot7024

		/// simpleSuite Version 1.0.1 \\\
					Notify
					
		Notify is a easy to use notification system
		which allows you to send notifications to players
		in your game.
		
		Usage:
		
		local notify = require(script.Parent.notify)
		notify.notify(player,"Title","Text","Color/Style",Duration)
		
		player should be a player, example: game.Players.Player1
		Title should be a string, example: "Hello,"
		Text should be a string, example: "world!"
		Color can be a string or table, example: "default" or "local Color = {["R"]=255,["G"]=0,["B"]=0}"
		Duration should be a number, example: 3
		
		Current Colors:
			alert - Red
			success - Green
			warning - Yellow
			info - Blue
			default - White
			NILColor - Gray(this color means the custom color sent has an NIL value)
			
		Custom Color System(Added In):
			1: To Display custom colors Just Send this through the regular Color varible, Start with this format:
				local Color = {
					["R"]=255, --Clamped To 0-255
					["G"]=0, --Clamped To 0-255
					["B"]=0, --Clamped To 0-255	
				}
			
			2: Then Send that into Notify just like before:
			   local notify = require(notify) -- Replace "notify" with the location of the module
			   local color = {["R"]=255, ["G"]=0, ["B"]=0,}
			   notify.notify(target,title,text,color,dur)

			3: You should Now have custom colors, 
				Side note!!: If it failed, the Notify Popups will use gray(Meaning You Did not send the Format Correctly)

		
		
		Notify is currently in a beta stage, more features will be added in the future.
		Please report any bugs you find to BotWot7024 (Discord Support can be found in our games, you will need RobloxVerfiy!)

--]]



Changes.lua
```
local module = {}
local TweenService = game.Workspace:WaitForChild("Services"):FindFirstChild("notify"):FindFirstChild("TweenService"):FindFirstChild("Event") -- Replace "notify" with the location of the module
local NVersion=script:WaitForChild("Version")
NVersion.Value="V1.03"
print("Started notify Module")
function insertContainer(p)
	if p == nil then
		local simpleSuite = Instance.new("ScreenGui")
		simpleSuite.Name = "simpleSuite"
		simpleSuite.DisplayOrder = 25
		simpleSuite.ZIndexBehavior = Enum.ZIndexBehavior.Sibling
		simpleSuite.Parent = game:GetService("StarterGui")

		local Notifications = Instance.new("Frame")
		Notifications.BackgroundTransparency = 1
		Notifications.Name = "Notifications"
		Notifications.Position = UDim2.new(0.815, 0, 0.08, 0)
		Notifications.Size = UDim2.new(0.175, 0, 0.9, 0)
		Notifications.Parent = simpleSuite

		local UIListLayout = Instance.new("UIListLayout")
		UIListLayout.HorizontalAlignment = Enum.HorizontalAlignment.Center
		UIListLayout.Padding = UDim.new(0.025, 0)
		UIListLayout.VerticalAlignment = Enum.VerticalAlignment.Bottom
		UIListLayout.Parent = Notifications
	else
		local simpleSuite = Instance.new("ScreenGui")
		simpleSuite.Name = "simpleSuite"
		simpleSuite.DisplayOrder = 25
		simpleSuite.ZIndexBehavior = Enum.ZIndexBehavior.Sibling
		simpleSuite.Parent = p:WaitForChild("PlayerGui")

		local Notifications = Instance.new("Frame")
		Notifications.BackgroundTransparency = 1
		Notifications.Name = "Notifications"
		Notifications.Position = UDim2.new(0.815, 0, 0.08, 0)
		Notifications.Size = UDim2.new(0.175, 0, 0.9, 0)
		Notifications.Parent = simpleSuite

		local UIListLayout = Instance.new("UIListLayout")
		UIListLayout.HorizontalAlignment = Enum.HorizontalAlignment.Center
		UIListLayout.Padding = UDim.new(0.025, 0)
		UIListLayout.VerticalAlignment = Enum.VerticalAlignment.Bottom
		UIListLayout.Parent = Notifications
	end
end

insertContainer()

--Preset colors
local colors = {
	NILColor = Color3.fromRGB(79, 79, 79),
	alert = Color3.fromRGB(255, 105, 105),
	success = Color3.fromRGB(105, 255, 105),
	warning = Color3.fromRGB(255, 255, 105),
	info = Color3.fromRGB(105, 105, 255),
	default = Color3.fromRGB(255, 255, 255)	
}

function module.notify(target,title,text,color,dur)

	if target:WaitForChild("PlayerGui"):FindFirstChild("simpleSuite") ~= nil then

		local duration

		if dur == nil then
			duration = 3
		else if dur ~= nil and dur >= 0.1 then
				duration = tonumber(dur)
			end
		end
		--Rescripted by botwot7024 (Below this line)
		if title ~= nil and text ~= nil and color ~= nil then
			local selectedColor = colors.default
			if type(color)=="string" then--Only here for backwards compatibility(old format)
				if string.lower(color) == "alert" or string.lower(color) == "success" or string.lower(color) == "warning" or string.lower(color) == "info" or string.lower(color) == "default" then
					if string.lower(color) == "alert" then
						selectedColor = colors.alert
					elseif string.lower(color) == "success" then
						selectedColor = colors.success
					elseif string.lower(color) == "warning" then
						selectedColor = colors.warning
					elseif string.lower(color) == "info" then
						selectedColor = colors.info
					elseif string.lower(color) == "default" then
						selectedColor = colors.default
					end
				else--if String but not table or know preset then default to NIL color
					selectedColor = colors.NILColor
				end	
			elseif type(color)=="table" then--If new format then
				if color.R==nil or color.G==nil or color.B==nil then--The New DefaultColor System
					selectedColor = colors.NILColor
				else
					selectedColor = Color3.fromRGB(math.clamp(color.R*255,0,255), math.clamp(color.G*255,0,255), math.clamp(color.B*255,0,255))
				end
			else--if no color defined at all default to "Default"
				selectedColor = colors.default
			end
			--Rescripted by botwot7024 (Above this line)
			local notification_number = 1


			if game.Players:FindFirstChild(target.Name) then
				local player = game.Players:FindFirstChild(target.Name)

				for _,v in pairs(player.PlayerGui.simpleSuite.Notifications:GetChildren()) do
					notification_number = notification_number +1
				end


				local Notification1 = Instance.new("Frame")
				Notification1.BackgroundColor3 = Color3.fromRGB(25, 25, 25)
				Notification1.BorderSizePixel = 0
				Notification1.Name = "Notification"..notification_number
				Notification1.Size = UDim2.new(0, 0, 0, 0)
				Notification1.Parent = player.PlayerGui:WaitForChild("simpleSuite").Notifications

				local Color = Instance.new("Frame")
				Color.BackgroundColor3 = selectedColor
				Color.BorderSizePixel = 0
				Color.Name = "Color"
				Color.Size = UDim2.new(0.025, 0, 1, 0)
				Color.Parent = Notification1

				local Content = Instance.new("Frame")
				Content.BackgroundTransparency = 1
				Content.Name = "Content"
				Content.Position = UDim2.new(0.025, 0, 0, 0)
				Content.Size = UDim2.new(0.975, 0, 1, 0)
				Content.Parent = Notification1

				local Title = Instance.new("TextLabel")
				Title.BackgroundTransparency = 1
				Title.Font = Enum.Font.SourceSansSemibold
				Title.Name = "Title"
				Title.Size = UDim2.new(1, 0, 0.4, 0)
				Title.Text = title
				Title.TextColor3 = Color3.fromRGB(255, 255, 255)
				Title.TextScaled = true
				Title.TextSize = 14
				Title.TextWrapped = true
				Title.Parent = Content

				local Text = Instance.new("TextLabel")
				Text.BackgroundTransparency = 1
				Text.Font = Enum.Font.SourceSans
				Text.Name = "Text"
				Text.Position = UDim2.new(0, 0, 0.2999997, 0)
				Text.Size = UDim2.new(1, 0, 0.7, 0)
				Text.Text = text
				Text.TextColor3 = Color3.fromRGB(255, 255, 255)
				Text.TextScaled = true
				Text.TextSize = 14
				Text.TextWrapped = true
				Text.Parent = Content

				--Rescripted by botwot7024
				if script:FindFirstChild("TweenService") then
					local Time=os.clock()
					print("Tweening 1 at "..Time)
					TweenService:Fire(Notification1, UDim2.new(1, 0, 0.1, 0), duration, UDim2.new(0, 0, 0, 0), 1.1, Time)
					print("Tweening 2 at "..os.clock())
				else
					Notification1:TweenSize(UDim2.new(1, 0, 0.1, 0))
					wait(duration)
					
					Notification1:TweenSize(UDim2.new(0, 0, 0, 0))
					wait(1.1)
					
					Notification1:Destroy()
				end
				--Rescripted by botwot7024
			end
		end
	else
		insertContainer(target)
		wait(0.05)
		module.notify(target,title,text,color,dur)
	end

end

return module
```
